crates/shielded_token/src/masp/shielded_wallet.rs

```diff
@@ -679,6 +644,45 @@ pub trait ShieldedApi<U: ShieldedUtils + MaybeSend + MaybeSync>:
         }
     }
 
+    /// Determine if using the current note would actually bring us closer to
+    /// our target. Returns the contribution of the current note to the
+    /// target if so.
+    #[allow(async_fn_in_trait)]
+    async fn is_amount_required(
+        &mut self,
+        client: &(impl Client + Sync),
+        src: ValueSum<(MaspDigitPos, Address), Change>,
+        dest: ValueSum<(MaspDigitPos, Address), Change>,
+        delta: I128Sum,
+    ) -> Option<ValueSum<(MaspDigitPos, Address), Change>> {
```

Comment:
Could `ValueSum<(MaspDigitPos, Address), Change>` possibly be changed to `ValueSum<(MaspDigitPos, Address), i128>` here?